indra/llwindow/llwindowsdl.cpp

```diff
@@ -654,6 +654,11 @@ BOOL LLWindowSDL::createContext(int x, int y, int width, int height, int bits, B
         SDL_SetHint( std::get<0>(hint), std::get<1>(hint));
     }
 
+#if LL_LINUX
+    // Enable IME on XWayland
+    setenv("XMODIFIERS", std::string("@im=fcitx;ibus").c_str(), 1);
```

Comment:
This should usually be set in via your DE environment/startup and thus globally and there should be no need for any application to overwrite it.

If it absolutely cannot be set for just the viewer, the best place is the viewer startup shell script.

Reply:
I agree, putting this into `indra/newview/linux_tools/wrapper.sh` would be a better solution

Reply:
I see.
I'll move there.